swaybar/ipc.c

```diff
@@ -306,18 +306,20 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 				ws->name = strdup(json_object_get_string(name));
 				ws->label = strdup(ws->name);
 				// ws->num will be -1 if workspace name doesn't begin with int.
-				size_t len_offset = ws->num == -1 ? 0 : numlen(ws->num);
-				if (bar->config->strip_workspace_name) {
-					free(ws->label);
-					ws->label = malloc(len_offset * sizeof(char));
-					ws->label[len_offset - 1] = '\0';
-					strncpy(ws->label, ws->name, len_offset);
-				} else if (bar->config->strip_workspace_numbers) {
-					len_offset += ws->label[len_offset] == ':';
-					if (strlen(ws->name) > len_offset) {
+				if(ws->num != -1) {
+					size_t len_offset = numlen(ws->num);
+					if (bar->config->strip_workspace_name) {
 						free(ws->label);
-						// Strip number prefix [1-?:] using len_offset.
-						ws->label = strdup(ws->name + len_offset);
+						ws->label = malloc(len_offset + 1 * sizeof(char));
+						ws->label[len_offset] = '\0';
+						strncpy(ws->label, ws->name, len_offset);
+					} else if (bar->config->strip_workspace_numbers) {
+						len_offset += ws->label[len_offset] == ':';
+						if (strlen(ws->name) > len_offset) {
+							free(ws->label);
+							// Strip number prefix [1-?:] using len_offset.
+							ws->label = strdup(ws->name + len_offset);
+						}
```

Comment:
If `strlen(ws->name) == len_offset` this should just strip the last colon, matching the behaviour of `strip_workspace_name`.

Reply:
Can you clarify further? Changing it to a `==` could cause problems because `len_offset` will always be one less than `strlen(ws->name)`.

Reply:
"1:" gets displayed as "1" if `strip_workspace_number` is enabled (though not if both are disabled).

Reply:
But you said this is how it was on i3, no?

>i3 also strips the colon if the name is something like 1:. I don't think this does that, though admittedly I haven't actually tested it.